controllers/topics/getSubscribeableTopic.js

```diff
@@ -20,32 +20,49 @@ module.exports = async (req, res) => {
     ORDER BY Topics.sort ASC
     LIMIT 10`;
 
-  const followedOnboardingTopicQueryResult = await sequelize.query(followedOnboardingTopicQuery, {
-    replacements: {
-      userId
-    }
-  });
-
   const topicHistoryQuery = `
     SELECT 
       DISTINCT name 
     FROM user_topic_history UserTopicsHistory 
     INNER JOIN topics Topics 
     ON UserTopicsHistory.topic_id = Topics.topic_id 
     WHERE UserTopicsHistory.user_id = :userId`;
-  const topicHistoryQueryResult = await sequelize.query(topicHistoryQuery, {
+
+  const topicHistoryQueryOperation = sequelize.query(topicHistoryQuery, {
+    replacements: {
+      userId
+    }
+  });
+
+  const followedOnboardingTopicQueryOperation = sequelize.query(followedOnboardingTopicQuery, {
     replacements: {
       userId
     }
   });
 
+  const hasUserPostedQueryOperation = Post.findAll({
+    where: {
+      author_user_id: userId
+    },
+    limit: 1,
+    raw: true
+  });
+
+  const [followedOnboardingTopicQueryResult, topicHistoryQueryResult, hasUserPosted] =
+    await Promise.all([
+      followedOnboardingTopicQueryOperation,
+      topicHistoryQueryOperation,
+      hasUserPostedQueryOperation
+    ]);
+
   res.json({
     status: 'success',
     code: 200,
     message: 'Success get topic user',
     data: {
       topics: followedOnboardingTopicQueryResult[0],
-      history: topicHistoryQueryResult[0]
+      history: topicHistoryQueryResult[0],
+      hasUserPosted: hasUserPosted?.length > 0
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are significant and improve the performance of the function by running the three queries concurrently using `Promise.all`. The addition of the `hasUserPostedQueryOperation` query is also correct and checks if the user has posted any content. The result of this query is then correctly added to the response object. However, there's a potential issue with error handling. If one of the promises fails, the other ones will still try to execute which might lead to unhandled promise rejections. Consider wrapping the `Promise.all` call in a try-catch block to handle any errors that might occur during the execution of the promises.

```diff
-  const [followedOnboardingTopicQueryResult, topicHistoryQueryResult, hasUserPosted] =
-    await Promise.all([
-      followedOnboardingTopicQueryOperation,
-      topicHistoryQueryOperation,
-      hasUserPostedQueryOperation
-    ]);
+  let followedOnboardingTopicQueryResult, topicHistoryQueryResult, hasUserPosted;
+  try {
+    [followedOnboardingTopicQueryResult, topicHistoryQueryResult, hasUserPosted] =
+      await Promise.all([
+        followedOnboardingTopicQueryOperation,
+        topicHistoryQueryOperation,
+        hasUserPostedQueryOperation
+      ]);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({
+      status: 'error',
+      code: 500,
+      message: 'An error occurred while fetching data'
+    });
+    return;
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->